miden/tests/integration/operations/stack_ops.rs

```diff
@@ -0,0 +1,35 @@
+use crate::build_op_test;
+use rand_utils::rand_value;
+
+// STACK OPERATIONS TESTS
+// ================================================================================================
+
+#[test]
+fn swapdw() {
+    let asm_op = "swapdw";
+
+    // --- simple case ----------------------------------------------------------------------------
+    let test = build_op_test!(
+        asm_op,
+        &[16, 15, 14, 13, 12, 11, 10, 9, 8, 7, 6, 5, 4, 3, 2, 1]
+    );
+    test.expect_stack(&[9, 10, 11, 12, 13, 14, 15, 16, 1, 2, 3, 4, 5, 6, 7, 8]);
+
+    // --- random values --------------------------------------------------------------------------
+    let mut vec: Vec<u64> = Vec::with_capacity(16);
+    for _ in 0..vec.capacity() {
+        vec.push(rand_value::<u64>());
```

Comment:
nit: i'd probably avoid a variable with name `vec`. Maybe call it `values` or something similar?